### ISSUE TITLE
DNSSEC fails with susepaste.org

### ISSUE THREAD

**Issue by wrosenauer** [PowerDNS/pdns#5827]
- Program: Recursor
 - Issue type: Bug report

### Short description
The recursor fails to validate/resolve the host susepaste.org using DNSSEC.

### Environment
 - Operating system: openSUSE Leap 42.3
 - Software version: 4.1.0-rc1
 - Software source: http://download.opensuse.org/repositories/home:/wrosenauer:/devel/

### Steps to reproduce
1. dnssec=validate
2. dig susepaste.org

### Expected behaviour
successful resolving the host

### Actual behaviour
SERVFAIL

**Comment by rgacogne** [comment]
Confirmed, what's happening is that we try to use ns5.gransy.com to resolve susepaste.org. We want to compute the validation state of ns5.gransy.com, so we request ns5.gransy.com|DS from ns1.anycastdns.cz. To get the address of ns1.anycastdns.cz we go to ns3.gransy.com..
I think it's now clear where we are headed.. and we finally hit:
```
Sending SERVFAIL to 127.0.0.1 during resolve of 'susepaste.org' because: More than 80 (max-recursion-depth) levels of recursion needed while resolving ns5.gransy.com
```

We don't have this issue in 4.0.x because we don't validate during the resolution process but after. The best option would probably be to disable validation of "infra" queries.

**Comment by rgacogne** [comment]
This is fixed by #5835.

**Comment by aerique** [comment]
#5835 has been merged into master.
